racksdb/web/ui/src/views/DatacentersView.vue

```diff
@@ -11,13 +11,67 @@ import { ref, onMounted } from 'vue'
 import type { Ref } from 'vue'
 import type { Datacenter } from '@/composables/RacksDBAPI'
 import BreadCrumbs from '@/components/BreadCrumbs.vue'
+import L from 'leaflet'
 import ComboBox from '@/components/ComboBox.vue'
+import { useRouter } from 'vue-router'
 
 const http = useHttp()
 const racksDBAPI = useRacksDBAPI(http)
 const datacenters: Ref<Array<Datacenter>> = ref([])
+const map = ref()
+const mapContainer = ref()
+const router = useRouter()
+const longitudes: Array<number> = []
+const latitudes: Array<number> = []
 async function getDatacenters() {
   datacenters.value = await racksDBAPI.datacenters()
+
+  // Average coordinates for correct map adjustment
+  datacenters.value.forEach((datacenter) => {
+    longitudes.push(datacenter.longitude)
+    latitudes.push(datacenter.latitude)
+  })
+
+  const averageLongitude =
+    longitudes.reduce((accumulator, currentValue) => accumulator + currentValue) / longitudes.length
+  const averageLatitude =
+    latitudes.reduce((accumulator, currentValue) => accumulator + currentValue) / latitudes.length
+
+  // Configuration of the map
+  map.value = L.map(mapContainer.value).setView([averageLatitude, averageLongitude], 5)
+  L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    maxZoom: 19,
+    attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+  }).addTo(map.value)
+
+  // Add a marker on a map if the datacenter has coordinates
+  datacenters.value.forEach((datacenter) => {
+    const datacenterRoute = router.resolve({
+      name: 'datacenterdetails',
+      params: { name: datacenter.name }
+    })
+    if (datacenter.latitude && datacenter.longitude) {
+      const purpleIcon = L.icon({
+        iconUrl: '/assets/racksdb-marker.svg',
```

Comment:
Do we really need the starting `/` in this URL? Relative URL would be compatible with deployments in subdirectories.

Reply:
No the starting `/` wasn't needed, I removed it.